libsession/src/main/java/org/session/libsession/snode/SnodeAPI.kt

```diff
@@ -277,28 +296,39 @@ object SnodeAPI {
         }
     }
 
-    fun getRawMessages(snode: Snode, publicKey: String): RawResponsePromise {
-//        val userED25519KeyPair = MessagingModuleConfiguration.shared.getUserED25519KeyPair() ?: return Promise.ofFail(Error.NoKeyPair)
+    fun getRawMessages(snode: Snode, publicKey: String, requiresAuth: Boolean = true, namespace: Int = 0): RawResponsePromise {
+        val userED25519KeyPair = MessagingModuleConfiguration.shared.getUserED25519KeyPair() ?: return Promise.ofFail(Error.NoKeyPair)
         // Get last message hash
-        val lastHashValue = database.getLastMessageHashValue(snode, publicKey) ?: ""
+        val lastHashValue = database.getLastMessageHashValue(snode, publicKey, namespace) ?: ""
+        val parameters = mutableMapOf<String,Any>(
+            "pubKey" to publicKey,
+            "last_hash" to lastHashValue,
+        )
         // Construct signature
-//        val timestamp = Date().time + SnodeAPI.clockOffset
-//        val ed25519PublicKey = userED25519KeyPair.publicKey.asHexString
-//        val verificationData = "retrieve$timestamp".toByteArray()
-//        val signature = ByteArray(Sign.BYTES)
-//        try {
-//            sodium.cryptoSignDetached(signature, verificationData, verificationData.size.toLong(), userED25519KeyPair.secretKey.asBytes)
-//        } catch (exception: Exception) {
-//            return Promise.ofFail(Error.SigningFailed)
-//        }
```

Comment:
Should we just delete these commented lines as we can always retrieve from git history if needed again?

Reply:
oh that was the old code, the new code makes use of this once more :)